api/server/router/network/network_routes.go

```diff
@@ -0,0 +1,248 @@
+		Warning: warning,
+	})
+}
+
+func (n *networkRouter) postNetworkConnect(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	var connect types.NetworkConnect
+	if err := httputils.ParseForm(r); err != nil {
+		return err
+	}
+
+	if err := httputils.CheckForJSON(r); err != nil {
+		return err
+	}
+
+	if err := json.NewDecoder(r.Body).Decode(&connect); err != nil {
+		return err
+	}
+
+	nw, err := findNetwork(n.netController, vars["id"], byID)
+	if err != nil {
+		return err
+	}
+
+	container, err := n.daemon.Get(connect.Container)
+	if err != nil {
+		return fmt.Errorf("invalid container %s : %v", container, err)
+	}
+	return container.ConnectToNetwork(nw.Name())
+}
+
+func (n *networkRouter) postNetworkDisconnect(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	var connect types.NetworkConnect
+	if err := httputils.ParseForm(r); err != nil {
+		return err
+	}
+
+	if err := httputils.CheckForJSON(r); err != nil {
+		return err
+	}
+
+	if err := json.NewDecoder(r.Body).Decode(&connect); err != nil {
+		return err
+	}
+
+	nw, err := findNetwork(n.netController, vars["id"], byID)
+	if err != nil {
+		return err
+	}
+
+	container, err := n.daemon.Get(connect.Container)
+	if err != nil {
+		return fmt.Errorf("invalid container %s : %v", container, err)
+	}
+	return container.DisconnectFromNetwork(nw)
+}
+
+func (n *networkRouter) deleteNetwork(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	if err := httputils.ParseForm(r); err != nil {
+		return err
+	}
+
+	nw, err := findNetwork(n.netController, vars["id"], byID)
+	if err != nil {
+		return err
+	}
+
+	return nw.Delete()
+}
+
+func findNetwork(c libnetwork.NetworkController, s string, by int) (libnetwork.Network, error) {
+	switch by {
+	case byID:
+		return c.NetworkByID(s)
+	case byName:
+		if s == "" {
+			s = c.Config().Daemon.DefaultNetwork
+		}
+		return c.NetworkByName(s)
+	}
+	return nil, errors.New("unexpected selector for network search")
+}
+
+func buildNetworkResource(nw libnetwork.Network) *types.NetworkResource {
+	r := &types.NetworkResource{}
+	if nw != nil {
+		r.Name = nw.Name()
+		r.ID = nw.ID()
+		r.Driver = nw.Type()
+		r.Containers = make(map[string]types.EndpointResource)
+		epl := nw.Endpoints()
+		for _, e := range epl {
+			sb := e.Info().Sandbox()
+			if sb == nil {
+				continue
+			}
+
+			er := types.EndpointResource{}
+			er.EndpointID = e.ID()
+			if iface := e.Info().Iface(); iface != nil {
+				if mac := iface.MacAddress(); mac != nil {
+					er.MacAddress = mac.String()
+				}
+				if ip := iface.Address(); len(ip.IP) > 0 {
+					er.IPv4Address = (&ip).String()
+				}
+
+				if ipv6 := iface.AddressIPv6(); len(ipv6.IP) > 0 {
+					er.IPv6Address = (&ipv6).String()
+				}
+			}
+			r.Containers[sb.ContainerID()] = er
+		}
+	}
+	return r
+}
+
+func processCreateDefaults(c libnetwork.NetworkController, n *types.NetworkCreate) {
```

Comment:
That function is a bit meh :-/  It's also used in a single place. Suggestion: can we rely on a `func newNetworkCreate() *types.NetworkCreate` to returns a default-initialized instance (with the `Driver` and options maps nicely instantiated), and just decode the JSON "over-it" and let it override whatever is specified as part of the JSON payload?